libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java

```diff
@@ -85,7 +85,8 @@ public class XContentParserTests extends OpenSearchTestCase {
         () -> randomAlphaOfLengthBetween(1, SmileXContent.DEFAULT_MAX_STRING_LEN),
         /* YAML parser limitation */
         XContentType.YAML,
-        () -> randomRealisticUnicodeOfCodepointLengthBetween(1, YamlXContent.DEFAULT_CODEPOINT_LIMIT)
+        /* limit to 75% of the limit, difficult to get the exact size of the content right */
```

Comment:
The YAML constraints apply to whole document, not just a value of particular key (and it is also in codepoints), it is difficult to get the limit exact, so relaxing constraints a bit.